pkg/macros/macros.go

```diff
@@ -93,6 +93,15 @@ func TimeInterval(query *sqlds.Query, args []string) (string, error) {
 	return fmt.Sprintf("toStartOfInterval(toDateTime(%s), INTERVAL %d second)", args[0], int(seconds)), nil
 }
 
+func TimeIntervalMs(query *sqlds.Query, args []string) (string, error) {
+	if len(args) != 1 {
+		return "", fmt.Errorf("%w: expected 1 argument, received %d", sqlds.ErrorBadArgumentCount, len(args))
+	}
+
+	milliseconds := math.Max(float64(query.Interval.Milliseconds()), 1)
+	return fmt.Sprintf("toStartOfInterval(toDateTime(%s), INTERVAL %d millisecond)", args[0], int(milliseconds)), nil
```

Comment:
Maybe to [toDateTime64](https://clickhouse.com/docs/en/sql-reference/functions/type-conversion-functions#todatetime64)  if you need sub-second precision? What scale would you set, then?

Reply:
Indeed, I missed that. If we want milliseconds to keep it consistent with other macros then I assume `toDateTime64(3, %s)`?

Reply:
sounds good